File: NoteS/controllers/PublicTagsController.cs
using Microsoft.AspNetCore.Mvc;
using NoteS.Attributes;
using NoteS.exceptions;
using NoteS.models.dto;
using NoteS.models.entity;
using NoteS.models.mappers;
using NoteS.services;
using NoteS.tools;
using Swashbuckle.AspNetCore.Annotations;

namespace NoteS.controllers;

[ApiController]
[Route("api/public/{accountName}")]
[ProducesResponseType(StatusCodes.Status403Forbidden)]
[ProducesResponseType(StatusCodes.Status401Unauthorized)]
public class PublicTagsController(
    TagInformationService tagInformationService,
    AccountRegisterService register,
    TagEditService tagEditService,
    UniversalDtoMapper um)
    : GeneralPreconditionController(register)
{
    [HttpGet("notes/{pathNote}/tags")]
    [KeycloakAuthorize(Policies.READ_NOTES)]
    [SwaggerOperation(Description = "Теги конкретной заметки")]
    public List<TagResponseDto> Tags([FromQuery] AccName accountName,
        [FromQuery] NotePath pathNote)
    {
        Check(accountName);
        var tags = tagInformationService.Tags(pathNote, accountName);
        return um.Of(tags);
    }

    [HttpPost("tags")]
    [KeycloakAuthorize(Policies.CREATE_NOTES)]
    [SwaggerOperation(Description = "Создание тега")]
    public CreatedResult CreateTag([FromQuery] AccName accountName,
        [FromBody] CreateTagRequestDto createTagRequestDto)
    {
        Check(accountName);
        TagResponseDto note = tagEditService.Create(accountName, createTagRequestDto);
        return Created(Url.Action("Tags", "PublicTags",
            new { accountName.AccountName }, Request.Scheme), note);
    }

    [HttpGet("tags")]
    [KeycloakAuthorize(Policies.READ_NOTES)]
    [SwaggerOperation(Description = "Список тегов пользователя")]
    public List<TagResponseDto> Tags([FromQuery] AccName accountName)
    {
        Check(accountName);

        return um.Of(tagInformationService.Tags(accountName));
    }

    [HttpDelete("notes/{pathNote}/tags/{tagName}")]
    [KeycloakAuthorize(Policies.READ_NOTES, Policies.DELETE_NOTES)]
    [SwaggerOperation(Description = "Удаление тега для заметки")]
    public NoContentResult DelTag([FromQuery] AccName accountName,
        [FromQuery] NotePath pathNote,
        [FromQuery] TagNameRequestDto delete)
    {
        Check(accountName);

        tagEditService.Delete(pathNote, accountName, delete);
        return NoContent();
    }

    [HttpPost("notes/{pathNote}/tags")]
    [KeycloakAuthorize(Policies.READ_NOTES, Policies.CREATE_NOTES)]
    [SwaggerOperation(Description = "Добавить тег к заметке")]
    public ActionResult AddTag([FromQuery] AccName accountName,
        [FromQuery] NotePath pathNote,
        [FromBody] AddTagRequestDto createTagRequestDto)
    {
        Check(accountName);
        var isCreate = tagEditService.Add(pathNote, accountName, createTagRequestDto);
        return isCreate ? Created() : Ok();
    }
}

File: javaBack/ResourceServer/src/main/java/com/notes/controllers/PublicTagsController.java
package com.notes.controllers;

import com.notes.models.dto.account.AccName;
import com.notes.models.dto.note.NotePath;
import com.notes.models.dto.tag.AddTagRequestDto;
import com.notes.models.dto.tag.CreateTagRequestDto;
import com.notes.models.dto.tag.TagNameRequestDto;
import com.notes.models.dto.tag.TagResponseDto;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.MediaType;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/api/private/{accountName}/notes")
@RequiredArgsConstructor
public class PublicTagsController {
   private final TagInformationService tagInformationService;
   private final AccountRegisterService register;
   private final TagEditService tagEditService;


   @GetMapping(path = "/notes/{pathNote}/tags", headers = "content-type=application/json", consumes = MediaType.APPLICATION_JSON_VALUE)
   @PreAuthorize("hasAnyAuthority('READ_NOTES')")
   public List<TagResponseDto> Tags(@Valid @PathVariable AccName accountName,
                                    @Valid @PathVariable NotePath pathNote) {
//      Check(accountName);
//      var tags = tagInformationService.Tags(pathNote, accountName);
//      return um.Of(tags);
   }

   @PostMapping(path = "/tags", headers = "content-type=application/json", consumes = MediaType.APPLICATION_JSON_VALUE)
   @PreAuthorize("hasAnyAuthority('READ_NOTES')")
   public CreatedResult CreateTag(@Valid @PathVariable AccName accountName,
                                  @Valid @RequestBody CreateTagRequestDto createTagRequestDto) {
//      Check(accountName);
//      TagResponseDto note = tagEditService.Create(accountName, createTagRequestDto);
//      return Created(Url.Action("Tags", "PublicTags",
//              new { accountName.AccountName }, Request.Scheme), note);
   }


   @GetMapping(path = "/tags", headers = "content-type=application/json", consumes = MediaType.APPLICATION_JSON_VALUE)
   @PreAuthorize("hasAnyAuthority('READ_NOTES')")
   public List<TagResponseDto> Tags(@Valid @PathVariable AccName accountName) {
//      Check(accountName);
//
//      return um.Of(tagInformationService.Tags(accountName));
   }

   @DeleteMapping(path = "/notes/{pathNote}/tags/{tagName}", headers = "content-type=application/json", consumes = MediaType.APPLICATION_JSON_VALUE)
   @PreAuthorize("hasAnyAuthority('READ_NOTES')")
   public NoContentResult DelTag(@Valid @PathVariable AccName accountName,
                                 @Valid @PathVariable NotePath pathNote,
                                 @Valid @PathVariable TagNameRequestDto tagName) {
//      Check(accountName);
//
//      tagEditService.Delete(pathNote, accountName, delete);
//      return NoContent();
   }

   @PostMapping(path = "/notes/{pathNote}/tags", headers = "content-type=application/json", consumes = MediaType.APPLICATION_JSON_VALUE)
   @PreAuthorize("hasAnyAuthority('READ_NOTES')")
   public ActionResult AddTag(@Valid @PathVariable AccName accountName,
                              @Valid @PathVariable NotePath pathNote,
                              @Valid @RequestBody AddTagRequestDto createTagRequestDto) {
//      Check(accountName);
//      var isCreate = tagEditService.Add(pathNote, accountName, createTagRequestDto);
//      return isCreate ? Created() : Ok();
   }
}
